ingest/ingest_integration_test.go

```diff
@@ -0,0 +1,1908 @@
+// Copyright (c) 2023 IndyKite
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ingest_test
+
+import (
+	"context"
+	"encoding/json"
+	"os"
+	"time"
+
+	"github.com/grpc-ecosystem/go-grpc-middleware/v2/interceptors/retry"
+	"google.golang.org/protobuf/encoding/protojson"
+
+	ingestpb "github.com/indykite/indykite-sdk-go/gen/indykite/ingest/v1beta2"
+	integration "github.com/indykite/indykite-sdk-go/test"
+
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+	. "github.com/onsi/gomega/gstruct"
+)
+
+var _ = Describe("Ingestion", func() {
+	Describe("IngestNode", func() {
+		It("UpsertNode", func() {
+			var err error
+			var jsonp = protojson.MarshalOptions{
+				Multiline:       true,
+				EmitUnpopulated: true,
+			}
+			ingestClient, err := integration.InitConfigIngest()
+			Expect(err).To(Succeed())
+
+			record, externalID := integration.CreateRecordIndividual(os.Getenv("TENANT_ID"), "Employee")
+			resp, err := ingestClient.IngestRecord(
+				context.Background(),
+				record,
+				retry.WithMax(2),
+			)
+
+			Expect(err).To(Succeed())
+			Expect(resp).To(PointTo(MatchFields(IgnoreExtras, Fields{
+				"RecordId": Not(BeNil()),
+				"Error":    BeNil(),
+				"Info": PointTo(MatchFields(IgnoreExtras, Fields{
+					"Changes": ContainElement(PointTo(MatchFields(IgnoreExtras, Fields{
+						"Id":       Not(BeEmpty()),
+						"DataType": Equal(ingestpb.Change_DATA_TYPE_DIGITAL_TWIN),
+					}))),
+				})),
+			})))
+
+			var response ingestpb.IngestRecordResponse
+			err = json.Unmarshal([]byte(jsonp.Format(resp)), &response)
+			Expect(err).To(Succeed())
```

Comment:
1. Why you marshal and unmarshal something? This doesn't make sense for me.
2. You cannot use `json` package with Proto messages.

Again, not writing comment everywhere